src/lib.rs

```diff
@@ -1853,10 +1853,20 @@ mod shard_tests {
             }
 
             // Check the unsorted read
+            assert_eq!(n_items, UnsortedShardReader::<T1>::len(&[tmp.path()])?);
             let unsorted_reader = UnsortedShardReader::<T1>::open(tmp.path());
             let all_items_res: Result<Vec<_>, Error> = unsorted_reader.collect();
             let all_items = all_items_res?;
             assert!(set_compare(&true_items, &all_items));
+
+            // Check that the skip_lazy feature produces the expected results.
+            let mut unsorted_reader_skip = UnsortedShardReader::<T1>::open(tmp.path());
+            let to_skip = (disk_chunk_size * 3) + 1;
```

Comment:
maybe scramble this value somehow to get different boundaries, or try a few different  carefully chosen skip values here? You could skip to a file boundary, a chunk boundary, or the middle of a chunk and ideally the tests would hit all 3 cases?

Reply:
I ultimately didn't merge the cellranger patch that made use of this feature, so I'm a bit on the fence about merging it at all at this point.

Reply:
Looking into this again now.

Reply:
I enhanced the tests by writing two shard files of size n_items, to test the set reading logic. I expanded the skip tests to include these offsets:
```
            check_unsorted_skip(0)?;
            check_unsorted_skip(1)?;
            check_unsorted_skip(disk_chunk_size)?;
            check_unsorted_skip((disk_chunk_size * 3) + 1)?;
            check_unsorted_skip(n_items)?; // skip entire first file
            check_unsorted_skip(n_items + 1)?; // skip entire first file plus next item
            check_unsorted_skip(n_items * 2)?; // skip everything
```